libraries/core_libs/network/src/tarcap/packets_handlers/transaction_packet_handler.cpp

```diff
@@ -24,31 +24,34 @@ inline void TransactionPacketHandler::process(const PacketData &packet_data, con
   std::string received_transactions;
   const auto transaction_count = packet_data.rlp_.itemCount();
 
-  SharedTransactions transactions;
+  std::vector<std::pair<std::shared_ptr<Transaction>, TransactionStatus>> transactions;
   transactions.reserve(transaction_count);
 
   for (size_t tx_idx = 0; tx_idx < transaction_count; tx_idx++) {
-    std::shared_ptr<Transaction> transaction;
+    std::pair<std::shared_ptr<Transaction>, TransactionStatus> transaction;
 
     try {
-      transaction = std::make_shared<Transaction>(packet_data.rlp_[tx_idx].data().toBytes());
+      transaction.first = std::make_shared<Transaction>(packet_data.rlp_[tx_idx].data().toBytes());
     } catch (const Transaction::InvalidSignature &e) {
       throw MaliciousPeerException("Unable to parse transaction: " + std::string(e.what()));
     }
 
     if (dag_blk_mgr_) [[likely]] {  // ONLY FOR TESTING
-      if (trx_mgr_->isTransactionKnown(transaction->getHash())) {
+      transaction.second = TransactionStatus::Verified;
+      if (trx_mgr_->isTransactionKnown(transaction.first->getHash())) {
         continue;
       }
-      if (const auto [is_valid, reason] = trx_mgr_->verifyTransaction(transaction); !is_valid) {
-        std::ostringstream err_msg;
-        err_msg << "Transaction " << transaction->getHash() << " validation failed: " << reason;
 
+      auto verification = trx_mgr_->verifyTransaction(transaction.first);
+      if (verification.first == TransactionStatus::Invalid) {
+        std::ostringstream err_msg;
+        err_msg << "Transaction " << transaction.first->getHash() << " validation failed: " << verification.second;
         throw MaliciousPeerException(err_msg.str());
       }
+      transaction.second = verification.first;
     }
-    received_transactions += transaction->getHash().abridged() + " ";
-    peer->markTransactionAsKnown(transaction->getHash());
+    received_transactions += transaction.first->getHash().abridged() + " ";
+    peer->markTransactionAsKnown(transaction.first->getHash());
     transactions.push_back(std::move(transaction));
```

Comment:
same as https://github.com/Taraxa-project/taraxa-node/pull/1714/files#r873469852 maybe you should insert status here not before as do not care and it complicates everything with first and second

Reply:
Fixed